app/containers/markdown/index.tsx

```diff
@@ -48,307 +28,37 @@ interface IMarkdownProps {
-
-				hardBreak: this.renderBreak,
-				thematicBreak: this.renderBreak,
-				softBreak: this.renderBreak,
-
-				htmlBlock: this.renderText,
-				htmlInline: this.renderText,
-
-				table: this.renderTable,
-				table_row: this.renderTableRow,
-				table_cell: this.renderTableCell
-			},
-			renderParagraphsInLists: true
-		});
-
-	get isNewMarkdown(): boolean {
-		const { md, enableMessageParser } = this.props;
-		return (enableMessageParser ?? true) && !!md;
-	}
-
-	renderText = ({ context, literal }: { context: []; literal: string }) => {
-		const { numberOfLines } = this.props;
-		const defaultStyle = [this.isMessageContainsOnlyEmoji ? styles.textBig : {}, ...context.map(type => styles[type])];
-		return (
-			<Text accessibilityLabel={literal} style={[styles.text, defaultStyle]} numberOfLines={numberOfLines}>
-				{literal}
-			</Text>
-		);
-	};
-
-	renderCodeInline = ({ literal }: TLiteral) => {
-		const { theme, style = [] } = this.props;
-		return (
-			<Text
-				style={[
-					{
-						...styles.codeInline,
-						color: themes[theme!].fontDefault,
-						backgroundColor: themes[theme!].surfaceNeutral,
-						borderColor: themes[theme!].surfaceNeutral
-					},
-					...style
-				]}>
-				{literal}
-			</Text>
-		);
-	};
-
-	renderCodeBlock = ({ literal }: TLiteral) => {
-		const { theme, style = [] } = this.props;
-		return (
-			<Text
-				style={[
-					{
-						...styles.codeBlock,
-						backgroundColor: themes[theme!].surfaceNeutral,
-						borderColor: themes[theme!].strokeLight,
-						color: themes[theme!].fontDefault
-					},
-					...style
-				]}>
-				{literal}
-			</Text>
-		);
-	};
-
-	renderBreak = () => {
-		const { tmid } = this.props;
-		return <Text>{tmid ? ' ' : '\n'}</Text>;
-	};
-
-	renderParagraph = ({ children }: any) => {
-		const { numberOfLines, style = [], theme, msg } = this.props;
-		if (!children || children.length === 0) {
-			return null;
-		}
-		if (msg && this.isMessageContainsOnlyEmoji) {
-			return <Text>{children}</Text>;
-		}
-		return (
-			<Text style={[styles.text, { color: themes[theme!].fontDefault }, ...style]} numberOfLines={numberOfLines}>
-				{children}
-			</Text>
-		);
-	};
-
-	renderLink = ({ children, href }: any) => {
-		const { theme, onLinkPress } = this.props;
+	if (!isTranslated) {
+		const tokens = md ?? parse(msg);
 		return (
-			<MarkdownLink link={href} theme={theme!} onLinkPress={onLinkPress}>
-				{children}
-			</MarkdownLink>
-		);
-	};
-
-	renderHashtag = ({ hashtag }: { hashtag: string }) => {
-		const { channels, navToRoomInfo, style } = this.props;
-		return <MarkdownHashtag hashtag={hashtag} channels={channels} navToRoomInfo={navToRoomInfo} style={style} />;
-	};
-
-	renderAtMention = ({ mentionName }: { mentionName: string }) => {
-		const { username = '', mentions, navToRoomInfo, useRealName, style } = this.props;
-		return (
-			<MarkdownAtMention
-				mentions={mentions}
-				mention={mentionName}
-				useRealName={useRealName}
+			<Body
```

Comment:
I had to open the code to understand what's `Body`.
While it works, I think we should put its content inside here, because that's a relevant part of how our markdown works.

Instead of 

```jsx
return <Body />
```

We would do

```jsx
return (
	<MarkdownContext.Provider
		value={{
			mentions,
			channels,
			useRealName,
			username,
			navToRoomInfo,
			getCustomEmoji,
			onLinkPress
		}}>
		{tokens?.map(block => {
			switch (block.type) {
				case 'BIG_EMOJI':
   ...
```